File: backend/app/src/main/java/com/yagubogu/place/repository/PlaceRepository.java
package com.yagubogu.place.repository;

import com.yagubogu.place.domain.Place;
import com.yagubogu.place.domain.PlaceCategory;
import java.util.List;
import org.springframework.data.jpa.repository.JpaRepository;
import org.springframework.stereotype.Repository;

@Repository
public interface PlaceRepository extends JpaRepository<Place, Long> {

    List<Place> findAllByStadiumIdAndCategory(Long stadiumId, PlaceCategory category);
}

File: mobile/composeApp/src/commonMain/kotlin/com/yagubogu/data/repository/place/PlaceRepository.kt
package com.yagubogu.data.repository.place

import com.yagubogu.data.dto.response.place.PlaceCategoryDto
import com.yagubogu.data.dto.response.place.PlaceDetailResponse
import com.yagubogu.data.dto.response.place.PlacesResponse

interface PlaceRepository {
    suspend fun getPlaces(
        stadiumId: Long,
        category: PlaceCategoryDto,
    ): Result<PlacesResponse>

    suspend fun getPlaceDetail(id: Long): Result<PlaceDetailResponse>
}
